### PR TITLE
Audio: DC block: Convert to module API

### PR DESCRIPTION
**Pull request by singalsu**
This patch converts the component to module API. The configuration blob handling is also changed to model handler.

### CODE REVIEW

File: src/audio/dcblock/dcblock.c [hunk 1]
```diff
@@ -65,255 +78,128 @@ static void dcblock_init_state(struct comp_data *cd)
...
+		goto err;
 	}
 
-	dev->state = COMP_STATE_READY;
-	return dev;
+	ret = comp_init_data_blob(cd->model_handler, bs, ipc_dcblock->data);
+	if (ret < 0) {
+		comp_err(dev, "dcblock_init(): comp_init_data_blob() failed with error: %d", ret);
+		comp_data_blob_handler_free(cd->model_handler);
+		goto err;
```
Comment (lyakh) [inline src/audio/dcblock/dcblock.c:113]: would be more future-proof to add a second error-handling label and move `comp_data_blob_handler_free()` there

**Review by lyakh — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@singalsu still draft ?

**Comment by singalsu** [comment]
> @singalsu still draft ?

Yes draft until I've tested it with IPC4 and tplg2. Should be verified by tomorrow.

File: src/audio/dcblock/dcblock.c (hunk 1 above)
Comment (singalsu) [reply src/audio/dcblock/dcblock.c:113]: Yep, done!

**Comment by singalsu** [comment]
> > @singalsu still draft ?
> 
> Yes draft until I've tested it with IPC4 and tplg2. Should be verified by tomorrow.

I got distracted by other work, but now DCblock is checked also with IPC4 system. Now a non-draft PR.

I will later make a PR for sof-hda-benchmark.conf to allow it to generate specific test topologies like one that I used for check: host copier --> dcblock --> dai copier, dai copier --> dcblock--> host copier.

File: src/audio/dcblock/dcblock.c [hunk 2]
```diff
@@ -65,255 +78,131 @@ static void dcblock_init_state(struct comp_data *cd)
...
  */
-static void dcblock_free(struct comp_dev *dev)
+static int dcblock_free(struct processing_module *mod)
 {
-	struct comp_data *cd = comp_get_drvdata(dev);
+	struct comp_data *cd = module_get_private_data(mod);
 
-	comp_info(dev, "dcblock_free()");
+	comp_info(mod->dev, "dcblock_free()");
+	comp_data_blob_handler_free(cd->model_handler);
```
Comment (cujomalainey) [inline src/audio/dcblock/dcblock.c:134]: nice catch

**Review by cujomalainey — APPROVED** [review]

**Comment by cujomalainey** [comment]
Looks like some errors in CI?
```
/zep_workspace/sof/src/audio/dcblock/dcblock.c:35:1: warning: symbol 'log_const_dcblock' was not declared. Should it be static?
/zep_workspace/sof/src/audio/dcblock/dcblock.c:38:1: warning: symbol 'dcblock_uuid' was not declared. Should it be static?
/zep_workspace/sof/src/audio/dcblock/dcblock.c:41:1: warning: symbol 'dcblock_tr' was not declared. Should it be static?
```

**Review by lgirdwood — APPROVED** [review]

**Review by btian1 — APPROVED** [review]

**Review by andrula-song — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Comment by kv2019i** [comment]
Hmm, @cujomalainey I can't see the build errors anymore, in which job this was? The stub and FW build runs seem clean now. The one sparse fail is a known one with a fix already in Zephyr main, so not related to this PR.

**Comment by kv2019i** [comment]
I'll wait one extra day for comments, and then proceed to merge.

**Pull request merged**